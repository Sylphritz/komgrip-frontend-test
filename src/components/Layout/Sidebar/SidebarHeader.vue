<template>
  <div class="sidenav-header">
    <i
      class="fas fa-times p-3 cursor-pointer text-secondary opacity-5 position-absolute end-0 top-0 d-xl-none"
      aria-hidden="true"
      id="iconSidenav"
      @click="toggleSidebar"
    ></i>
    <a class="navbar-brand m-0" href="/" target="_blank">
      <img src="/images/logo.png" class="navbar-brand-img h-100" alt="main_logo" />
      <span class="ms-1 font-weight-bold">Komgrip Technologies</span>
    </a>
  </div>
</template>

<script setup lang="ts">
import { useLayoutStore } from '@/stores/layout'

const layoutStore = useLayoutStore()
const { toggleSidebar } = layoutStore
</script>
