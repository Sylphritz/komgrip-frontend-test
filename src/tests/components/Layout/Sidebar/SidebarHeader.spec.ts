import { describe, it, expect, beforeEach } from 'vitest'

import { mount } from '@vue/test-utils'
import SidebarHeader from '@/components/Layout/Sidebar/SidebarHeader.vue'
import { createPinia, setActivePinia } from 'pinia'

describe('SidebarHeader', () => {
  beforeEach(() => {
    setActivePinia(createPinia())
  })

  it('matches the snapshot', () => {
    const wrapper = mount(SidebarHeader)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has the title of Komgrip Technologies', () => {
    const wrapper = mount(SidebarHeader)
    expect(wrapper.text()).toBe('Komgrip Technologies')
  })
})
