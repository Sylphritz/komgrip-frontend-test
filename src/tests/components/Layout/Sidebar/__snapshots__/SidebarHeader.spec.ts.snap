// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarHeader > matches the snapshot 1`] = `
<div
  class="sidenav-header"
>
  <i
    aria-hidden="true"
    class="fas fa-times p-3 cursor-pointer text-secondary opacity-5 position-absolute end-0 top-0 d-xl-none"
    id="iconSidenav"
  />
  <a
    class="navbar-brand m-0"
    href="/"
    target="_blank"
  >
    <img
      alt="main_logo"
      class="navbar-brand-img h-100"
      src="/images/logo.png"
    />
    <span
      class="ms-1 font-weight-bold"
    >
      Komgrip Technologies
    </span>
  </a>
</div>
`;
